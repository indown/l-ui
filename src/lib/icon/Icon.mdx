---
name: icon 
router: /icon
---

import { Playground, Props } from 'docz'
import Icon from './index'

# Icon

<Props of={Icon} />

## Basic usage

<Playground>
  <Icon type='wechat' />
  <Icon type='qq' />
</Playground>