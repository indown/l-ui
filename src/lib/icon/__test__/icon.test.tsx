import Icon from '../icon'
import React from 'react';
import renderer from 'react-test-renderer';
import { mount, render } from 'enzyme'

describe('<Icon />', () => {
  it('icon is a react component', () => {
    const component = renderer.create(
      <Icon type='qq' />
    );
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('icon be onClick', () => {
    const fn = jest.fn()
    const wrapper = mount(<Icon type='qq' onClick={fn}/>)
    wrapper.find('svg').simulate('click')
    expect(fn).toBeCalledTimes(1)
  })
  it('icon has this className', () => {
    const wrapper = render(<Icon className='test-classname' type='qq'/>)
    expect(wrapper.hasClass('test-classname'))
  })
})