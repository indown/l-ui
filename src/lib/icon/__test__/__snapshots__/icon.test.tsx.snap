// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> icon is a react component 1`] = `
<svg
  className="l-icon"
>
  <use
    xlinkHref="#qq"
  />
</svg>
`;
